packages/components/image-viewer/src/image-viewer.vue

```diff
@@ -83,19 +83,20 @@
           </div>
           <!-- CANVAS -->
           <div :class="ns.e('canvas')">
-            <template v-for="(url, i) in urlList" :key="i">
-              <img
-                v-if="i === activeIndex"
-                :ref="(el) => (imgRefs[i] = el as HTMLImageElement)"
-                :src="url"
-                :style="imgStyle"
-                :class="ns.e('img')"
-                :crossorigin="crossorigin"
-                @load="handleImgLoad"
-                @error="handleImgError"
-                @mousedown="handleMouseDown"
-              />
-            </template>
+            <slot v-if="loadError" name="viewer-error">
+              <div :class="ns.e('error')">{{ t('el.image.error') }}</div>
+            </slot>
+            <img
+              v-else
+              ref="imgRef"
+              :src="currentImg"
+              :style="imgStyle"
+              :class="ns.e('img')"
+              :crossorigin="crossorigin"
+              @load="handleImgLoad"
+              @error="handleImgError"
+              @mousedown="handleMouseDown"
+            />
```

Comment:
I think we can keep the original structure and pass the `url` and `index` to the slot `viewer-error`.